src/detect-lua.c

```diff
@@ -1951,64 +1780,39 @@ static int LuaMatchTest03a(void)
     StreamTcpInitConfig(true);
 
     DetectEngineCtx *de_ctx = DetectEngineCtxInit();
-    if (de_ctx == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(de_ctx);
     de_ctx->flags |= DE_QUIET;
 
     s = DetectEngineAppendSig(de_ctx, sig);
-    if (s == NULL) {
-        printf("sig parse failed: ");
-        goto end;
-    }
-
+    FAIL_IF_NULL(s);
     SigGroupBuild(de_ctx);
     DetectEngineThreadCtxInit(&th_v, (void *)de_ctx, (void *)&det_ctx);
 
     /* do detect for p1 */
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p1);
 
-    if ((PacketAlertCheck(p1, 1))) {
-        printf("sid 1 didn't match on p1 but should have: ");
-        goto end;
-    }
+    FAIL_IF(PacketAlertCheck(p1, 1));
 
     /* do detect for p2 */
     SigMatchSignatures(&th_v, de_ctx, det_ctx, p2);
-
-    if (!(PacketAlertCheck(p2, 1))) {
-        printf("sid 1 didn't match on p2 but should have: ");
-        goto end;
-    }
+    FAIL_IF_NOT(PacketAlertCheck(p2, 1));
 
     FlowVar *fv = FlowVarGet(&f, 1);
-    if (fv == NULL) {
-        printf("no flowvar: ");
-        goto end;
-    }
 
-    if (fv->data.fv_str.value_len != 1) {
-        printf("%u != %u: ", fv->data.fv_str.value_len, 1);
-        goto end;
-    }
+    FAIL_IF_NULL(fv);
 
-    if (memcmp(fv->data.fv_str.value, "2", 1) != 0) {
-        PrintRawDataFp(stdout, fv->data.fv_str.value, fv->data.fv_str.value_len);
+    FAIL_IF(fv->data.fv_str.value_len != 1);
 
-        printf("buffer mismatch: ");
-        goto end;
-    }
+    FAIL_IF(memcmp(fv->data.fv_str.value, "2", 1) != 0);
 
-    result = 1;
-end:
     if (de_ctx != NULL)
-        DetectEngineCtxFree(de_ctx);
+        DetectEngineCtxFree(de_ctx); // should this be kept ?
```

Comment:
Hi, sorry for taking so long to send a new pr.
I have a question about this section if I remove these two lines (1808,1809) half of the tests fail. I didn't quite understand why, did I made some mistake previous to this and it should be fix ? 

Thanks : )

Reply:
Hello, we understand that volunteer work can be like that, sometimes. If something is urgent, we would probably have kindly told you that someone on the team would be taking it over, and thanked you for your efforts up to that point. That said, thanks for getting back to it! :)

About your question, now:
I think my previous feedback was misleading, I apologize. We must still call `DetectEngineCtxFree(de_ctx)` and the following clean-up routines. We just don't need the `if (de_ctx != NULL)` check, as this part will have been covered already if we reach this point in the test. Does that make sense?